ProgrammaticUI/Controllers/HomeViewController.swift

```diff
@@ -65,6 +67,39 @@ class HomeViewController: UIViewController {
         navigationController?.pushViewController(insertDataVC, animated: true)
     }
 
+    private func loadData() {
+        guard let userEmail = Auth.auth().currentUser?.email else {
+            fatalError("User not authenticated.")
```

Comment:
This stops execution. Do we want to crash if `Auth.auth().currentUser?.email` is nil?

Reply:
Same in `getUsername(from:)`

Reply:
Realmente nunca llega a ejecutarse eso porque primero se valida el usuario. No deja saltar a la siguiente pantalla si aparece un login raro. 
Lo acabo de probar y no se rompe. lo único que salta por consola es: "The email address is badly formatted."
Si pongo ambos vacíos: "The password is invalid or the user does not have a password."

Entiendo que está mal hecho de esta manera pero en qué momento podría romperse sabiendo esto ? y cuál sería la mejor manera?  solo usando un if let y lanzando un print o incluso con el guard pero lanzando un UIAlert mostrando el error no ?

Reply:
In a production app, you could present the user with some visual feedback like an alert window. Also, I would log this in two different ways:

1. When building for production; log the failure event in Firebase and present the user with a warning.
2. When building for debugging; log the failure event locally.

Reply:
On the other topic you mentioned and to answer your question: this stops execution when `Auth.auth().currentUser` is `nil`.